cuda_core/cuda/core/experimental/_memoryview.pyx

```diff
@@ -74,15 +73,26 @@ cdef class StridedMemoryView:
         The pointer address (as Python `int`) to the **consumer** stream.
         Stream ordering will be properly established unless ``-1`` is passed.
     """
-    # TODO: switch to use Cython's cdef typing?
-    ptr: int = None
-    shape: tuple = None
-    strides: tuple = None  # in counts, not bytes
-    dtype: numpy.dtype = None
-    device_id: int = None  # -1 for CPU
-    is_device_accessible: bool = None
-    readonly: bool = None
-    exporting_obj: Any = None
+    cdef readonly:
+        intptr_t ptr
+        int device_id
+        bint is_device_accessible
+        bint readonly
+        object exporting_obj
+    
+    # If using dlpack, this is a strong reference to the result of 
+    # obj.__dlpack__() so we can lazily create shape and strides from 
+    # it later.  If using CAI, this is a reference to the source 
+    # `__cuda_array_interface__` object.
+    cdef object metadata
+
+    # The tensor object if has obj has __dlpack__, otherwise must be NULL
+    cdef DLTensor *dl_tensor
+        
+    # Memoized properties
+    cdef tuple _shape
+    cdef object _strides
```

Comment:
IIRC `self._strides = None` is allowed in Cython even if `_strides` is typed to tuple, so this might help readability (and perf too?)
```suggestion
    cdef tuple _strides
```